doc/kernel/usermode/usermode.rst

```diff
@@ -9,6 +9,176 @@ are defined, and how memory may be managed to support user mode threads.
+    access to read or write, depending on the semantics of the API.
+
+  - Use of kernel objects that are not in a proper initialization state.
+
+- We ensure the detection and safe handling of user mode stack overflows.
+
+- We prevent invoking system calls to functions excluded by the kernel
+  configuration.
+
+- We prevent disabling of or tampering with kernel-defined and hardware-
+  enforced memory protections.
+
+- We prevent re-entry from user to supervisor mode except through the kernel-
+  defined system calls and interrupt handlers.
+
+- We prevent the introduction of new executable code by user mode threads,
+  except to the extent to which this is supported by kernel system calls.
+
+We are specifically not protecting against the following attacks:
+
+- The kernel itself, and any threads that are executing in supervisor mode,
+  are assumed to be trusted.
+
+- The kernel build is assumed to be trusted. There is considerable build-time
+  logic for creating the tables of valid kernel objects, defining system calls,
+  and configuring interrupts. The .elf binary files that are worked with
+  during this process are all assumed to be trusted code.
+
+- We can't protect against mistakes made in memory domain configuration done in
+  kernel mode that exposes private kernel data structures to a user thread. RAM
+  for kernel objects should always be configured as supervisor-only.
+
+- It is possible to make top-level declarations of user mode threads and
+  assign them permissions to kernel objects. In general, all C and header
+  files that are part of the kernel build producing zephyr.elf are assumed to
+  be trusted.
+
+- We do not protect against denial of service attacks through thread CPU
+  starvation. Zephyr has no thread priority aging and a user thread of a
+  particular priority can starve all threads of lower priority, and also other
+  threads of the same priority if time-slicing is not enabled.
+
+- There are build-time defined limits on how many threads can be active
+  simultaneously, after which creation of new user threads will fail.
+
+- Stack overflows for threads running in supervisor mode may be caught,
+  but the integrity of the system cannot be guaranteed.
+
+High-level Policy Details
+=========================
+
+Broadly speaking, we accomplish these thread-level memory protection goals
+through the following mechanisms:
+
+- Any user thread will only have access to its own stack memory by default.
+  Access to any other RAM will need to be done on the thread's behalf through
+  system calls, or specifically granted by a supervisor thread using the
+  :ref:`memory_domain` APIs. Newly created threads inherit the memory domain
+  configuration of the parent. Threads may communicate with each other
+  by having shared membership of the same memory domains, or via kernel objects
+  such as semaphores and pipes.
+
+- If the optional Application Memory feature is enabled, all threads will have
```

Comment:
How about, 

    If the optional option:`CONFIG_APPLICATION_MEMORY` feature is enabled, ...